src/DynamoCoreWpf/ViewModels/Core/StateMachine.cs

```diff
@@ -129,13 +129,14 @@ internal void BeginDragSelection(Point2D mouseCursor)
             // Here each node in the selection is being recorded for undo right
             // before they get updated by the drag operation.
             // 
-            RecordSelectionForUndo();
+            //RecordSelectionForUndo();
             draggedNodes.Clear();
             foreach (ISelectable selectable in DynamoSelection.Instance.Selection)
             {
                 ILocatable locatable = selectable as ILocatable;
+                ModelBase nodeModel = selectable as ModelBase;
                 if (null != locatable)
-                    draggedNodes.Add(new DraggedNode(locatable, mouseCursor));
+                    draggedNodes.Add(new DraggedNode(locatable, mouseCursor, nodeModel.GUID));
```

Comment:
Maybe check null for both before referencing?

Reply:
I will remove this as a parameter and cast it in the constructor. Thanks!